### PR TITLE
Add version entry to gconsturct expected schema.

### PR DESCRIPTION
**Pull request by thvasilo**
*Issue #, if available:*
https://github.com/awslabs/graphstorm/issues/389

*Description of changes:*

Add a `version` entry to gconstruct specification.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by classicsong — APPROVED** [review]
LGTM

File: docs/source/configuration/configuration-gconstruction.rst [hunk 1]
```diff
@@ -101,10 +105,11 @@ An example
 ............
 Below shows an example that contains one node type and an edge type. For a real example, please refer to the :ref:`input JSON file <input-config>` used in the :ref:`Use Your Own Graphs Tutorial <use-own-data>`.
 
-.. code-block:: yaml
+.. code-block:: json
 
     {
-        nodes: [
+        "version": "gconstruct-0.1",
```
Comment (zheng-da) [inline docs/source/configuration/configuration-gconstruction.rst:111]: should we use something like `v0.1` as the version number?

File: python/graphstorm/gconstruct/construct_graph.py [hunk 2]
```diff
@@ -549,6 +549,10 @@ def process_edge_data(process_confs, node_id_map, arr_merger,
 def verify_confs(confs):
     """ Verify the configuration of the input data.
     """
+    if "version" not in confs:
```
[the remark this answers is not in the document]
Comment (thvasilo) [reply python/graphstorm/gconstruct/construct_graph.py:552]: In Gconstruct itself it's defined here for future use. If the config spec changes, we'll need to support previous versions if we want to keep backwards compatibility with existing spec files.

In GSProcessing we will use it as:

```python
if "version" in dataset_config_dict:
    # The config is one of gconstruct-with-version, neptuneml, or gsprocessing
    
    if dataset_config_dict["version"] == "gconstruct-1.0":
        # Handle gconstruct
    elif dataset_config_dict["version"] == "gsprocessing-1.0":
        # Handle gsprocessing
    elif dataset_config_dict["version"] == "v2.0":
        # Handle NeptuneML
else:
    # The config is legacy gconstruct without version
    # Handle gconstruct
```

File: docs/source/configuration/configuration-gconstruction.rst (hunk 1 above)
Comment (thvasilo) [reply docs/source/configuration/configuration-gconstruction.rst:111]: Sure, I can make this change.

**Review by zheng-da — APPROVED** [review]

**Pull request merged**